src/Lykke.Logs/SanitizingLog.cs

```diff
@@ -0,0 +1,195 @@
+    /// </summary>
+    public class SanitizingLog : ILog
+    {
+        private readonly ILog _log;
+        private readonly List<(Regex pattern, string replacement)> _patterns = new List<(Regex, string)>();
+
+        /// <summary>
+        /// Initializes new instance.
+        /// </summary>
+        /// <param name="log">Original log instance.</param>
+        public SanitizingLog(ILog log) => this._log = log ?? throw new System.ArgumentNullException(nameof(log));
+
+        /// <summary>
+        /// Adds sensitive pattern that should not be logged. Api keys, private keys and so on.
+        /// </summary>
+        /// <param name="pattern">Regex that should be replaced.</param>
+        /// <param name="replacement">Pattern replacement.</param>
+        public SanitizingLog AddSensitivePattern(Regex pattern, string replacement)
+        {
+            _patterns.Add((pattern, replacement));
+            return this;
+        }
+
+        /// <summary>
+        /// Replaces all patterns provided through <see cref="AddSensitivePattern(Regex, string)" /> with corresponding values.
+        /// </summary>
+        /// <param name="value">String to sanitize.</param>
+        /// <returns></returns>
+        public string Sanitize(string value) => _patterns.Aggregate(value, (a, p) => p.pattern.Replace(a, p.replacement));
+
+        IDisposable ILog.BeginScope(string scopeMessage)
+        {
+            return _log.BeginScope(scopeMessage);
+        }
+
+        bool ILog.IsEnabled(Microsoft.Extensions.Logging.LogLevel logLevel)
+        {
+            return _log.IsEnabled(logLevel);
+        }
+
+        void ILog.Log<TState>(Microsoft.Extensions.Logging.LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var sanitizedException = 
+                exception is SanitizingException ? exception : 
+                exception != null ? new SanitizingException(exception, Sanitize) :
+                null;
+
+            var sanitizedFormatter = formatter != null ?
+                new Func<LogEntryParameters, Exception, string>((s, e) => Sanitize(formatter(state, exception))) :
+                null;
+
+            var sanitizedState = new LogEntryParameters(
+                state.AppName,
+                state.AppVersion,
+                state.EnvInfo,
+                Sanitize(state.CallerFilePath),
+                Sanitize(state.Process),
+                state.CallerLineNumber,
+                Sanitize(state.Message),
+                Sanitize(state.Context),
+                state.Moment);
+
+            _log.Log(logLevel, eventId, sanitizedState, sanitizedException, sanitizedFormatter);
+        }
+
+        Task ILog.WriteErrorAsync(string component, string process, string context, Exception exception, DateTime? dateTime)
+        {
+            return _log.WriteErrorAsync(
```

Comment:
Legacy methods WriteXXXAsync implementations in Log, use ILog.Log method in the end. Thus sanitazing will be made twice. Exceptions are WriteMonitorAsync overloads they use IHealthNotifier. So you need to pass through parameters for all WriteXXXAsync methods except WriteMonitorAsync, It's not clear. But these methods are obsolete and will be removed soon

Reply:
Decided to avoid assumptions about `ILog` implementations and use less productive but more honest version.

Reply:
NOTE: there is no double-sanitizing actually due to using aggregation but not inheritance, so that obsolete methods don't call generic `Log()` method of `SanitizingLog` instance.